File: core/session/src/remediation.rs
//! Structured agent-facing guidance — mirrors Python's
//! `complier.runtime.remediation.StructuredMessage`.

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct StructuredMessage {
    pub summary: String,
    #[serde(default)]
    pub details: Vec<String>,
    #[serde(default)]
    pub allowed_next_actions: Vec<String>,
}

impl StructuredMessage {
    pub fn new(summary: impl Into<String>) -> Self {
        Self {
            summary: summary.into(),
            details: Vec::new(),
            allowed_next_actions: Vec::new(),
        }
    }
}

File: src/complier/runtime/remediation.py
"""Helpers for constructing agent-facing remediation messages."""

from __future__ import annotations

from dataclasses import dataclass, field


@dataclass(slots=True)
class StructuredMessage:
    """Structured guidance returned to an agent after a blocked action."""

    summary: str
    details: list[str] = field(default_factory=list)
    allowed_next_actions: list[str] = field(default_factory=list)
